src/input/label.component.ts

```diff
@@ -7,89 +7,104 @@ import {
-					'bx--text-input__invalid-icon': !textArea,
-					'bx--text-area__invalid-icon': textArea
+					'bx--label--disabled': disabled,
+					'bx--skeleton': skeleton,
+					'bx--label--inline': isInline,
+					'bx--label--inline--sm': (isInline && size === 'sm'),
+					'bx--label--inline--md': (isInline && size === 'md'),
+					'bx--label--inline--xl': (isInline && size === 'xl')
 				}">
-			</svg>
-			<svg
-				*ngIf="!invalid && warn"
-				ibmIcon="warning--alt--filled"
-				size="16"
-				class="bx--text-input__invalid-icon bx--text-input__invalid-icon--warning">
-			</svg>
-			<ng-content select="input,textarea,div"></ng-content>
-		</div>
-		<div
-			*ngIf="!skeleton && helperText && !invalid && !warn"
-			class="bx--form__helper-text"
-			[ngClass]="{'bx--form__helper-text--disabled': disabled}">
-			<ng-container *ngIf="!isTemplate(helperText)">{{helperText}}</ng-container>
-			<ng-template *ngIf="isTemplate(helperText)" [ngTemplateOutlet]="helperText"></ng-template>
-		</div>
-		<div *ngIf="!warn && invalid" class="bx--form-requirement">
-			<ng-container *ngIf="!isTemplate(invalidText)">{{invalidText}}</ng-container>
-			<ng-template *ngIf="isTemplate(invalidText)" [ngTemplateOutlet]="invalidText"></ng-template>
+				<ng-content></ng-content>
+			</label>
+			<ng-template *ngIf="isInline"
+				[ngTemplateOutlet]="helperTextTemplate">
+			</ng-template>
 		</div>
-		<div *ngIf="!invalid && warn" class="bx--form-requirement">
-			<ng-container *ngIf="!isTemplate(warnText)">{{warnText}}</ng-container>
-			<ng-template *ngIf="isTemplate(warnText)" [ngTemplateOutlet]="warnText"></ng-template>
+		<div class="bx--text-input__field-outer-wrapper"
+			[ngClass]="{
+			'bx--text-input__field-outer-wrapper--inline': isInline
+			}">
+			<div
+				[class]="wrapperClass"
+				[ngClass]="{
+					'bx--text-input__field-wrapper--warning': isWarning
+				}"
+				[attr.data-invalid]="(invalid ? true : null)"
+				#wrapper>
+				<svg
+					*ngIf="!isWarning && invalid && !isReadonly"
+					ibmIcon="warning--filled"
+					size="16"
+					[ngClass]="{
+						'bx--text-input__invalid-icon': !textArea,
+						'bx--text-area__invalid-icon': textArea
+					}">
+				</svg>
+				<svg
+					*ngIf="!invalid && isWarning && !isReadonly"
+					ibmIcon="warning--alt--filled"
+					size="16"
+					class="bx--text-input__invalid-icon bx--text-input__invalid-icon--warning">
+				</svg>
+				<svg
+					*ngIf="isReadonly"
+					ibmIcon="edit--off"
+					size="16"
+					class="bx--text-input__readonly-icon">
+				</svg>
+				<ng-content select="input,textarea,div"></ng-content>
+			</div>
+			<ng-container *ngIf="!isInline"
+				[ngTemplateOutlet]="helperTextTemplate">
+			</ng-container>
 		</div>
+
+		<ng-template #helperTextTemplate>
+			<div *ngIf="!skeleton && helperText && ((!invalid && !isWarning) || isInline)"
+				class="bx--form__helper-text"
+				[ngClass]="{
+					'bx--form__helper-text--disabled': disabled,
+					'bx--form__helper-text--inline': inline
```

Comment:
I had missed adding this `--inline` class to helper text... 
- https://github.com/carbon-design-system/carbon/blob/v10/packages/react/src/components/TextInput/TextInput.js#L111-L114